### ISSUE TITLE
For NARGP, the kernel function of the high fidelity GP is probably wrong.

### ISSUE THREAD

**Issue by HHkazan** [IceLab-X/FidelityFusion#11]
For NARGP, the kernel function of the high fidelity GP is probably wrong. In this package, the kernel function of the high fidelity GP is still simple RFB, but not consider the LFGP's output. Besides, the optimizer to optimize the hyperparameters should restart several times to avoid local optima.

**Comment by wayXing** [comment]
Thank you for your feedback. We will start looking into this issue. 

"Besides, the optimizer to optimize the hyperparameters should restart several times to avoid local optima."
Do you mean that multiple optimizations with different initializations should be conducted in order to get good hyperparameters?

**Comment by HHkazan** [comment]
Yes. It seems to be one usual solution to avoid local optima in other GP packages, such as GPy. But it dosen't work well when the dimension of data is high. 
Besides, I'm trying if the novel stochastic optimization method can be used in this situation, such as the Whale optimization method proposed by SEYEDALI MIRJALILI.

**Comment by wayXing** [comment]
That is a perfect practice for real applications. We will incorporate this in our later version. 
But when making comparisons with other methods, this trick has to be applied to the methods otherwise it might seem unfair. If all methods are run without this trick and with multiple runs, it should also be a valid comparison.
